src/network/protocol.rs

```diff
@@ -697,11 +707,37 @@ impl<T: Config> UdpProtocol<T> {
         self.round_trip_time = millis - body.pong;
     }
 
+    /// Upon recveiving a `ChecksumReport`, add it to the checksum history
+    fn on_checksum_report(&mut self, body: &ChecksumReport) {
+        if self.last_added_checksum_frame < body.frame {
+            while self.checksum_history.len() > MAX_CHECKSUM_HISTORY_SIZE {
```

Comment:
I am not sure I understand why this is a while-loop. Shouldn't you be able to "forget" old checksums in a single function call?

Reply:
yeah this is correct. the while loop was a remnant of the Veqdeque implementation of the algorithm